index.js

```diff
@@ -3,11 +3,26 @@ module.exports = function (str, opts) {
 	str = str.toString();
 	opts = opts || {};
 
-	var preserve = !opts.all;
+	var comment = '';
 	var currentChar = '';
 	var insideString = false;
+	var preserveFilter;
+	var preserveImportant = (opts.preserve === false || opts.all === true) ? false : true;
 	var ret = '';
 
+	//use preserveFilter as an interface for opts.preserve function and RegExp values
+	if (opts !== undefined && opts.preserve !== undefined) {
+		if ((typeof opts.preserve).toLowerCase() === 'function') {
+			preserveImportant = false;
+			preserveFilter = opts.preserve;
+		} else if ((opts.preserve.constructor.name).toLowerCase() === 'regexp') {
+			preserveImportant = false;
+			preserveFilter = function (comment) {
+				return opts.preserve.test(comment);
+			};
+		}
+	}
```

Comment:
I hope I've understood correctly what you are suggesting. Apologies if I haven't.

I'm not assuming that `opts.preserve` is either a function or a `RegExp` (and nothing else) because it could also be a boolean. The outer conditional is satisfied if `opts.preserve === true` or `opts.preserve === false`. If `opts.preserve` is a boolean, `preserveFilter` should be `undefined`. 

Also, by explicitly checking for both a RegExp and a function, calling strip-css-comments with an invalid `opts.preserve` value such as `'false'` won't have a catastrophic effect. Instead, invalid configuration is ignored. I'm not sure if this is more desirable than throwing a validation error. 

To simplify, I've removed `if (opts.preserve !== undefined)` and am now just checking the type of `opts.preserve`.